### PR TITLE
Remove broken federation feature

### PR DESCRIPTION
**Pull request by mrnugget**
Only today did I learn about the "Federation" feature: https://docs.sourcegraph.com/admin/federation/public_repositories

It's supposed to redirect users from private instances to sourcegraph.com.

But it has been broken since at least a year: https://github.com/sourcegraph/sourcegraph/issues/38263#issuecomment-1205110145

It's also confusing.

Here's my proposal to remove it.

## Test plan

- N/A

### CODE REVIEW

**Review by sashaostrikov — APPROVED** [review]
The whole feature in under 70 lines? 😁 
UN:shipit: (un-ship it)

**Review by BolajiOlajide — APPROVED** [review]

File: schema/site.schema.json [hunk 1]
```diff
@@ -803,12 +803,6 @@
         ]
       ]
     },
-    "disablePublicRepoRedirects": {
```
Comment (eseliger) [inline schema/site.schema.json:806]: This value was in confdefaults so if we just remove this property it'll break validation for all existing instances. We should probably just deprecate it for now.

Comment (mrnugget) [reply schema/site.schema.json:806]: But we do allow additional keys, no?

Comment (mrnugget) [reply schema/site.schema.json:806]: https://github.com/sourcegraph/sourcegraph/blob/f45ef88ad88f7c4e55ec518ee09a95017817ebba/schema/site.schema.json#L8

**Comment by mrnugget** [comment]
Added changelog and only deprecated the value instead of removing it.

File: schema/site.schema.json [hunk 2]
```diff
@@ -804,7 +804,7 @@
       ]
     },
     "disablePublicRepoRedirects": {
-      "description": "Disable redirects to sourcegraph.com when visiting public repositories that can't exist on this server.",
+      "description": "DEPRECATED! Disable redirects to sourcegraph.com when visiting public repositories that can't exist on this server.",
```
Comment (eseliger) [inline schema/site.schema.json:807]: https://github.com/sourcegraph/sourcegraph/blob/852f39d043a0a09ba47a4c37c598dfd698941554/schema/site.schema.json#L57 Want to add such a message? I think the editor renders these

**Review by eseliger — APPROVED** [review]

File: schema/site.schema.json (hunk 2 above)
Comment (mrnugget) [reply schema/site.schema.json:807]: done

**Comment by mrnugget** [comment]
@sqs I merged this without waiting for approval. I think this feature being broken for over 1 year & no engineer here knowing about the feature at all, plus it just adding just enough complexity to make the code hard to understand is a good reason to remove it.

**Pull request merged**